datafusion/functions/src/datetime/date_part.rs

```diff
@@ -223,6 +233,30 @@ impl ScalarUDFImpl for DatePartFunc {
     }
 }
 
+fn is_integar_part(part: &str) -> bool {
+    let part = part_normalization(part);
+    matches!(
+        part.to_lowercase().as_str(),
+        "year"
+            | "month"
+            | "week"
+            | "day"
+            | "hour"
+            | "minute"
+            | "qtr"
+            | "quarter"
+            | "doy"
+            | "dow"
```

Comment:
second, millisecond, micro, nanos too?

Reply:
```rust
(secs as f64 + ((subsecs % 1_000_000_000) as f64 / 1_000_000_000_f64))
                        * sf
```

Probably we still need f64 for seconds

Reply:
i didn't know `extract(second ..` returns seconds with fraction. In PostgreSQL it does so, in e.g. Trino and Snowflake it does not

```
postgres=# select extract(second from now());
  extract
-----------
 29.009019
```

```
trino> select extract(second from now());
 _col0
-------
    46
```

so yes, keep seconds with float for now

Reply:
extract is float correct, returning the fractions

Reply:
filed https://github.com/apache/datafusion/issues/13482 for this

Reply:
@jayzhan211 what about millisecond, micro, nano?